xtm/xtm_api.h

```diff
@@ -0,0 +1,159 @@
+#pragma once
+/*
+ * Copyright 2010-2021, Tarantool AUTHORS, please see AUTHORS file.
+ *
+ * Redistribution and use in source and binary forms, with or
+ * without modification, are permitted provided that the following
+ * conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above
+ *    copyright notice, this list of conditions and the
+ *    following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above
+ *    copyright notice, this list of conditions and the following
+ *    disclaimer in the documentation and/or other materials
+ *    provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY <COPYRIGHT HOLDER> ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * <COPYRIGHT HOLDER> OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+ * BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+#include <stdlib.h>
+
+#if defined(__cplusplus)
+extern "C" {
+#endif /* defined(__cplusplus) */
+
+/** \cond public */
+
+/**
+ * Opaque struct, that represent one-directional, one-reader-one-writer
+ * queue implementation with eventloop integration ability.
+ */
+struct xtm_queue;
+
+/**
+ * Typedef for often used function pointer, used in dispatch/invoke
+ * xtm pattern.
+ */
+typedef void (*fun_for_dispatch)(void*);
+
+/**
+ * Create instance of struct xtm_queue.
+ * @param[in] size - queue size, must be power of two
+ * @retval pointer to new xtm_queue or NULL in case of error.
+ *
+ */
+struct xtm_queue *
+xtm_queue_new(unsigned size);
+
+/**
+ * Free queue and close its internal fds.
+ * @param[in] queue - xtm_queue to delete
+ * @retval 0 on success. Otherwise -1 with errno set appropriately
+ *         (as in close(2), since it implies close of fds).
+ */
+int
+xtm_queue_delete(struct xtm_queue *queue);
+
+/**
+ * Notify queue consumer.
+ * @param[in] queue - xtm_queue to notify
+ * @retval 0 on success. Otherwise -1 with errno set appropriately
+ *         (as in write(2), since it implies write to internal fd).
+ */
+int
+xtm_msg_notify(struct xtm_queue *queue);
```

Comment:
If Mons likes how the names are short, you could find a compromise. For example, `xtq` or `xtmq` - then `queue` becomes redundant. `xtmq_new()`, `xtmq_delete()`, `xtmq_...`. `xtm` alone does not work, because as you said in README, it is "inter thread (xt) messaging (m)". Word "messaging" != "queue", not even closely related.